src/cml.js

```diff
@@ -164,26 +165,43 @@ class CML {
     const triggerSha = await this.triggerSha();
     const {
       commitSha: inCommitSha = triggerSha,
-      rmWatermark,
-      update,
+      markdownFile,
       pr,
       publish,
       publishUrl,
-      markdownFile,
       report: testReport,
+      rmWatermark,
+      triggerFile,
+      update,
       watch,
-      triggerFile
+      watermarkTitle
     } = opts;
 
+    const drv = this.getDriver();
+
     const commitSha =
       (await this.revParse({ ref: inCommitSha })) || inCommitSha;
 
     if (rmWatermark && update)
       throw new Error('watermarks are mandatory for updateable comments');
 
+    // Create the watermark.
+    const genWatermark = ({ label, workflow, run } = opts) => {
+      // Replace {workflow} and {run} placeholders in label with actual values.
+      label = label
+        .replace('{workflow}', workflow)
+        .replace('{run}', run)
+        .trim();
+      const title = `CML watermark ${label}`.trim();
+      return `![](${WATERMARK_IMAGE} "${title}")`;
+    };
```

Comment:
```suggestion
    const genWatermark = ({ label, workflow, run } = opts) => {
      const title = label
        .replace('{workflow}', workflow)
        .replace('{run}', run)
        .trim();
      return `![](${WATERMARK_IMAGE} "CML watermark ${title}")`;
    };
```

Reply:
I think the comment do not add a lot of value... can be seen fast in the code

Also I prefer to not modify passed values. In this case works but js has funny things like [call-by-sharing](http://en.wikipedia.org/wiki/Evaluation_strategy#Call_by_sharing) and its safer to just consider the pass as reference

I think that you do not need to trim twice 🙂

Reply:
Its also a bit strange label vs title

Reply:
We need the second trim, so that if `label` is empty, the resulting watermark will be `![](_img_ "CML watermark")`, not `![](_img_ "CML watermark ")` (the extra space at the end of the watermark title)

Reply:
Ok you do not need the first trim then

Reply:
dropped it - thanks!